### PR TITLE
lftp 4.7.5 (new formula)

### PR DESCRIPTION
**Pull request by fxcoudert**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a pull request to revive lftp, now that it builds again on macOS (almost unpatched). There was a pull request already, but its author has stopped updating it in response to reviews (https://github.com/Homebrew/homebrew-core/pull/7371) so I am opening this one.

### CODE REVIEW

File: Formula/lftp.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  homepage "https://lftp.tech"
+  url "https://lftp.yar.ru/ftp/lftp-4.7.5.tar.bz2"
+  mirror "ftp://ftp.st.ryukoku.ac.jp/pub/network/ftp/lftp/lftp-4.7.5.tar.bz2"
+  sha256 "90f3cbc827534c3b3a391a2dd8b39cc981ac4991fa24b6f90e2008ccc0a5207d"
+
+  depends_on "readline"
+  depends_on "openssl"
+  depends_on "libidn"
+
+  # On Yosemite, the system's openssl gets chosen over ours, and it is too old
```
Comment (woodruffw) [inline Formula/lftp.rb]: This sounds like a build system bug that should probably be fixed upstream.

File: Formula/lftp.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}",
+                          "--with-openssl=#{Formula["openssl"].opt_prefix}",
+                          "--with-readline=#{Formula["readline"].opt_prefix}",
+                          "--with-libidn=#{Formula["libidn"].opt_prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/lftp", "-c", "open ftp://mirrors.kernel.org; ls"
```
Comment (woodruffw) [inline Formula/lftp.rb]: I believe `mirrors.kernel.org` is being shut down later this year; it might be good to look for a replacement to test with.

Comment (fxcoudert) [inline Formula/lftp.rb]: Replaced with GNU ftp

File: Formula/lftp.rb (hunk 1 above)
Comment (fxcoudert) [inline Formula/lftp.rb]: I have opened an upstream issue: https://github.com/lavv17/lftp/issues/317

Comment (woodruffw) [inline Formula/lftp.rb]: Thanks for reporting, and adding a note to the formula. I don't think it'll be a blocker here, but it'll be nice to remove that line eventually :smile:

**Comment by woodruffw** [comment]
This looks good to me. I'll leave it open a bit to give other maintainers a chance to review.

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**